pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -191,29 +211,148 @@ func (r *ReconcileUserAccount) ensureIdentity(logger logr.Logger, userAcc *toolc
+	return errs.Wrapf(err, format, args...)
 }
 
-// wrapErrorWithStatusUpdate wraps the error and set the user account status to "provisioning". If the update failed then log the error.
-func (r *ReconcileUserAccount) wrapErrorWithStatusUpdate(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount, err error, format string, args ...interface{}) error {
-	if err == nil {
-		return nil
+// updateStatusConditions updates user account status conditions with the new conditions
+func (r *ReconcileUserAccount) updateStatusConditions(userAcc *toolchainv1alpha1.UserAccount, newConditions ...toolchainv1alpha1.Condition) error {
+	var atLeastOneUpdated bool
+	conditions := userAcc.Status.Conditions
+	var updated bool
+	for _, condition := range newConditions {
+		conditions, updated = AddOrUpdateStatusCondition(conditions, condition)
+		atLeastOneUpdated = atLeastOneUpdated || updated
 	}
-	if err := r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error()); err != nil {
-		logger.Error(err, "status update failed")
+	if !atLeastOneUpdated {
+		// Nothing changed
+		return nil
 	}
-	return errs.Wrapf(err, format, args...)
+	userAcc.Status.Conditions = conditions
+	return r.client.Status().Update(context.TODO(), userAcc)
+}
+
+func (r *ReconcileUserAccount) setStatusUserCreationFailed(userAcc *toolchainv1alpha1.UserAccount, message string) error {
+	reason := unableToCreateUserReason
+	return r.updateStatusConditions(
+		userAcc,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.UserAccountUserNotReady,
+			Status:  apiv1.ConditionTrue,
+			Reason:  reason,
+			Message: message,
+		},
+		toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.UserAccountReady,
+			Status: apiv1.ConditionFalse,
+			Reason: reason,
+		})
+}
+
+func (r *ReconcileUserAccount) setStatusIdentityCreationFailed(userAcc *toolchainv1alpha1.UserAccount, message string) error {
+	reason := unableToCreateIdentityReason
+	return r.updateStatusConditions(
+		userAcc,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.UserAccountIdentityNotReady,
+			Status:  apiv1.ConditionTrue,
+			Reason:  reason,
+			Message: message,
+		},
+		toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.UserAccountReady,
+			Status: apiv1.ConditionFalse,
+			Reason: reason,
+		})
+}
+
+func (r *ReconcileUserAccount) setStatusMappingCreationFailed(userAcc *toolchainv1alpha1.UserAccount, message string) error {
+	reason := unableToCreateMappingReason
+	return r.updateStatusConditions(
+		userAcc,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.UserAccountUserIdentityMappingNotReady,
+			Status:  apiv1.ConditionTrue,
+			Reason:  reason,
+			Message: message,
+		},
+		toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.UserAccountReady,
+			Status: apiv1.ConditionFalse,
+			Reason: reason,
+		})
+}
+
+func (r *ReconcileUserAccount) setStatusUserCreationOK(userAcc *toolchainv1alpha1.UserAccount) error {
```

Comment:
I believe that this method and the following two can be merged into one:
```go 
func (r *ReconcileUserAccount) setStatusOK(userAcc *toolchainv1alpha1.UserAccount, conType toolchainv1alpha1.ConditionType) error {
	return r.updateStatusConditions(
		userAcc,
		toolchainv1alpha1.Condition{
			Type:   conType,
			Status: apiv1.ConditionFalse,
		})
}
```

Reply:
While the current implementation is more verbose (I agree) the usage IMO is a bit clearer:

```
setStatusUserCreationOK(userAcc)
```
vs.
```
setStatusOK(userAcc, toolchainv1alpha1.UserAccountUserNotReady)
```